packages/es-dev-server/src/middleware/polyfills-loader.ts

```diff
@@ -1,44 +1,48 @@
-/** @typedef {import('./polyfills-loader-types').PolyfillsLoaderMiddlewareConfig} PolyfillsLoaderMiddlewareConfig */
-/** @typedef {import('polyfills-loader').GeneratedFile} GeneratedFile */
-
-/**
- * @typedef {object} IndexHTMLData
- * @property {string} indexHTML
- * @property {string} lastModified
- * @property {GeneratedFile[]} inlineScripts
- */
-
 /* eslint-disable no-console */
 import path from 'path';
 import { URLSearchParams } from 'url';
-import { injectPolyfillsLoader } from '../utils/inject-polyfills-loader.js';
+import { Middleware } from 'koa';
+import { GeneratedFile } from 'polyfills-loader';
+
+import { injectPolyfillsLoader } from '../utils/inject-polyfills-loader';
 import {
   isIndexHTMLResponse,
   getBodyAsString,
   toBrowserPath,
   isInlineScript,
   RequestCancelledError,
   toFilePath,
-} from '../utils/utils.js';
-import { getUserAgentCompat } from '../utils/user-agent-compat.js';
+} from '../utils/utils';
+import { getUserAgentCompat } from '../utils/user-agent-compat';
+import { PolyfillsLoaderConfig } from '../utils/inject-polyfills-loader';
+import { TransformJs } from '../utils/compatibility-transform';
+
+interface IndexHTMLData {
+  indexHTML: string;
+  lastModified: string;
+  inlineScripts: GeneratedFile[];
+}
+
+export interface PolyfillsLoaderMiddlewareConfig {
+  compatibilityMode: string;
+  appIndex?: string;
+  rootDir: string;
+  polyfillsLoaderConfig?: Partial<PolyfillsLoaderConfig>;
+  transformJs: TransformJs;
+}
 
 /**
  * Creates middleware which injects polyfills and code into the served application which allows
  * it to run on legacy browsers.
- *
- * @param {PolyfillsLoaderMiddlewareConfig} cfg
  */
-export function createPolyfillsLoaderMiddleware(cfg) {
+export function createPolyfillsLoaderMiddleware(cfg: PolyfillsLoaderMiddlewareConfig): Middleware {
   // polyfills, keyed by url
-  /** @type {Map<string, string>} */
-  const polyfills = new Map();
+  const polyfills = new Map<string, string>();
```

Comment:
maybe should be `Map<Polyfill, string>` where `enum Polyfill {...}`

Reply:
or even better `Map<keyof PolyfillsConfig, string>`

Reply:
`keyof` is cool! But this is a mapping of the polyfill URL to the file content.